src/Symfony/Component/PropertyInfo/Util/PhpDocTypeHelper.php

```diff
@@ -108,9 +111,11 @@ private function createType(DocType $docType): ?Type
 
         if ('mixed[]' === $docTypeString) {
             $docTypeString = 'array';
+        } elseif ('array' !== $docTypeString && $docType instanceof Array_ && null === new \ReflectionProperty(Array_::class, 'keyType')->getValue($docType)) {
```

Comment:
`__toString()` was implemented in `Array_` in phpDocumentor/TypeResolver#274 slightly changing what is returned when the `$keyType` is `null`

@jaapio Do you think it might make sense to add a method that allows to determine if a key/value type was set?

Reply:
You mean: https://github.com/phpDocumentor/TypeResolver/blob/2.x/src%2FTypes%2FAbstractList.php#L48 ? 

This method will return the key set via the docblock. That's why it can return null. 
The getKey method will always return a type. Which is array-key by default when nothing was set.